lib/ffmpeg/dash/adaptation_set.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require_relative 'hls_class_methods'
+require_relative 'representation'
+
+module FFMPEG
+  module DASH
+    # Represents an Adaptation Set in a DASH manifest.
+    class AdaptationSet
+      include HLSClassMethods
+
+      attr_reader :manifest
+
+      def initialize(manifest, node)
+        @manifest = manifest
+        @node = node
+      end
+
+      # Returns the ID of the adaptation set.
+      #
+      # @return [Integer, nil] The ID of the adaptation set.
+      def id
+        @id ||= @node['id']&.to_i
+      end
+
+      # Returns the aspect ratio of the adaptation set.
+      #
+      # @return [String, nil] The pixel aspect ratio.
+      def par
+        @par ||= @node['par']
+      end
+
+      # Returns the language of the adaptation set.
+      #
+      # @return [String, nil] The language code (e.g., 'und', 'en', 'fr').
+      def lang
+        @lang ||= @node['lang']
+      end
+
+      # Returns the content type of the adaptation set.
+      #
+      # @return [String, nil] The content type.
+      def content_type
+        @content_type ||= @node['contentType']
+      end
+
+      # Returns the max width of the adaptation set.
+      #
+      # @return [Integer, nil] The maximum width in pixels.
+      def max_width
+        @max_width ||= @node['maxWidth']&.to_i
+      end
+
+      # Returns the max height of the adaptation set.
+      #
+      # @return [Integer, nil] The maximum height in pixels.
+      def max_height
+        @max_height ||= @node['maxHeight']&.to_i
+      end
+
+      # Returns the frame rate of the adaptation set.
+      #
+      # @return [Rational, nil] The frame rate as a Rational number.
+      def frame_rate
+        @frame_rate ||= @node['frameRate']&.to_r
+      end
+
+      # Returns the representations in the adaptation set.
+      #
+      # @return [Array<Representation>, nil] An array of Representation objects.
+      def representations
```

Comment:
why do we have the option to return `nil`? does it mean something else than a zero-length array?